### PR TITLE
[5.1] Deprecating RouterLegacy class

### PR DESCRIPTION
**Pull request by Hackwar**
### Summary of Changes
Since Joomla 3.3 we have the class-based routers and it is time to remove the support for the old function-based routers. This PR deprecates the legacy support class for the old routers. 

### Testing Instructions
Codereview

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [ ] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [ ] No documentation changes for manual.joomla.org needed

### CODE REVIEW

**Comment by brianteeman** [comment]
What is the policy now for when something can be removed?

File: libraries/src/Component/Router/RouterLegacy.php [hunk 1]
```diff
@@ -17,6 +17,9 @@
  * Default routing class for missing or legacy component routers
  *
  * @since  3.3
+ * @deprecated  5.1 will be removed in 6.0
```
Comment (laoneo) [inline libraries/src/Component/Router/RouterLegacy.php]:
```suggestion
 * @deprecated  5.1 will be removed in 7.0
```
It can't be removed before 7.0.

**Comment by Hackwar** [comment]
@brianteeman if something is deprecated in a major version, it can be removed 2 major versions later.

**Comment by brianteeman** [comment]
> @brianteeman if something is deprecated in a major version, it can be removed 2 major versions later.

Which is why ~I asked two weeks ago when you submitted this PR saying removal in 6.0

**Comment by Hackwar** [comment]
And the production department can deviate from that rule if they deem it necessary. I would have removed it earlier, but the department voted against this. So 7.0 it is.

**Comment by bembelimen** [comment]
Thx

**Pull request merged**